rustls/src/crypto/hmac.rs

```diff
@@ -45,11 +52,45 @@ impl Tag {
 
 impl Drop for Tag {
     fn drop(&mut self) {
-        self.buf.zeroize();
+        self.0.buf.zeroize();
     }
 }
 
 impl AsRef<[u8]> for Tag {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_ref()
+    }
+}
+
+/// A HMAC tag, stored as a value, that is non-secret.
+///
+/// A value of this type is not zeroized on drop.
+///
+/// A tag is "public" if it is published on the wire. For example, the
+/// `verify_data` field of TLS `Finished` messages are public (as they
+/// are published on the wire in TLS1.2, or sent encrypted under
+/// pre-authenticated secrets in TLS1.3).
+#[derive(Clone)]
+pub(crate) struct PublicTag {
+    buf: [u8; Tag::MAX_LEN],
+    used: usize,
+}
+
+impl PublicTag {
+    /// Build a tag by copying a byte slice.
+    ///
+    /// The slice can be up to [`Tag::MAX_LEN`] bytes in length.
+    pub(crate) fn new(bytes: &[u8]) -> Self {
+        let mut tag = Self {
+            buf: [0u8; Tag::MAX_LEN],
```

Comment:
nit: `0` vs `0u8`? 

_(Thinking along the lines of how we avoid explicit type annotations elsewhere, but know this is super nit-picky :laughing:)_

Reply:
Yeah, I tend to prefer relying on inference where possible because I think fewer tokens will generally be easier to consume.